Modules/System/MicrosoftGraph/src/MgAuthorization/MgGraphClientCredentials.Codeunit.al

```diff
@@ -0,0 +1,83 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+
+codeunit 9137 "MgGraph Client Credentials" implements "Microsoft Graph Authorization"
+{
+    Access = Internal;
+
+    var
+        BearerTxt: Label 'Bearer %1', Comment = '%1 - Token', Locked = true;
+        ClientCredentialsTokenAuthorityUrlTxt: Label 'https://login.microsoftonline.com/%1/oauth2/v2.0/token', Comment = '%1 = AAD tenant ID', Locked = true;
+        [NonDebuggable]
+        Scopes: List of [Text];
+        [NonDebuggable]
+        AadTenantId: Text;
+        [NonDebuggable]
+        AuthCodeErr: Text;
+        [NonDebuggable]
+        ClientId: Text;
+        [NonDebuggable]
+        ClientSecret: Text;
+
+    [NonDebuggable]
+    procedure SetParameters(NewAadTenantId: Text; NewClientId: Text; NewClientSecret: Text; NewScopes: List of [Text])
+    begin
+        AadTenantId := NewAadTenantId;
+        ClientId := NewClientId;
+        ClientSecret := NewClientSecret;
+        Scopes := NewScopes;
+    end;
+
+    [NonDebuggable]
+    procedure Authorize(var HttpRequestMessage: HttpRequestMessage);
+    var
+        Headers: HttpHeaders;
+    begin
+        HttpRequestMessage.GetHeaders(Headers);
+        Headers.Add('Authorization', StrSubstNo(BearerTxt, GetToken()));
+    end;
+
+    [NonDebuggable]
+    local procedure GetToken(): Text
+    var
+        [NonDebuggable]
```

Comment:
Just wondering, but if the procedure is already NonDebuggable, does adding that tag to a local variable as well have any effect?